application/services/stripe_client.py

```diff
@@ -51,15 +51,17 @@ def __init__(self, api_key):
 
     def f(self, func):
         def stripe_call(*args, **kwargs):
-            if current_user:
-                user_id = current_user.id
-
+            user_id = 'NOT_SET'
             try:
+                if current_user:
```

Comment:
functionality here is not to check for authentication but to get the current user id, if the user is not logged in it will be un-set.
The main reason for this check is so that the code can work outside of flask application context also. look at stripe_wrapper.py